### PR TITLE
#653: Added HSCAN command

### PR DESCRIPTION
**Pull request by manishrw**
## Changes:
- Added HSCAN command to iterate over hash fields and values
- Implemented cursor-based iteration and glob pattern matching
- Added unit tests
- Added integration tests
- Added doc for HSCAN command

Issue: #653

### CODE REVIEW

**Comment by manishrw** [comment]
@JyotinderSingh  @lucifercr07 Can you review it and approve if it looks good?

**Comment by manishrw** [comment]
@JyotinderSingh @lucifercr07 Can you review and approve this PR?

File: integration_tests/commands/async/hscan_test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import (
+	"testing"
+
+	"gotest.tools/v3/assert"
+)
+
+var THREE int64 = 3
+var FOUR int64 = 4
+
+func TestHSCAN(t *testing.T) {
```
Comment (JyotinderSingh) [inline integration_tests/commands/async/hscan_test.go:12]: Please add a test case where a new cursor value is returned, then use that cursor value to get the remaining elements.

File: internal/eval/eval.go [hunk 2]
```diff
@@ -3238,6 +3240,83 @@ func evalHDEL(args []string, store *dstore.Store) []byte {
...
+				}
+				count = parsedCount
+			}
+		}
+	}
+
+	keys := make([]string, 0, len(hashMap))
+	for k := range hashMap {
+		keys = append(keys, k)
+	}
```
Comment (JyotinderSingh) [inline internal/eval/eval.go:3288]: This makes our method have O(n) complexity (which is different from redis, which offers O(1)), please add a comment here mentioning the same. Could you also update this commands doc page in the docs directory with this information?

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for adding support for HSCAN command @manishrw! Left a few comments.

File: integration_tests/commands/async/hscan_test.go (hunk 1 above)
Comment (manishrw) [reply integration_tests/commands/async/hscan_test.go:12]: have added the test case

File: internal/eval/eval.go (hunk 2 above)
Comment (manishrw) [reply internal/eval/eval.go:3290]: done.

**Comment by manishrw** [comment]
@JyotinderSingh I've addressed the comments.

**Review by JyotinderSingh — APPROVED** [review]
Thanks for adding support for `HSCAN` command and updating the docs, @manishrw! Changes look good and well-tested.

**Pull request merged**